pulumi/index.ts

```diff
@@ -0,0 +1,97 @@
+import * as pulumi from "@pulumi/pulumi";
+import * as eks from "@pulumi/eks";
+import * as k8s from "@pulumi/kubernetes";
+import * as awsx from "@pulumi/awsx";
+
+import * as fs from 'fs';
+import * as YAML from 'yaml'
+
+const repo = new awsx.ecr.Repository("tezos-k8s");
+
+// Manual step: create a pulumi_values.yaml in the top level dir
+// with mkchain command:
+//   mkchain pulumi
+//
+
+let chainName = process.env.CHAIN_NAME || "pulumi";
+
+const helmValuesFile = fs.readFileSync(chainName + '_values.yaml', 'utf8')
+const helmValues = YAML.parse(helmValuesFile)
+
+const nginxIngressHelmValues_file = fs.readFileSync('nginx_ingress_values.yaml', 'utf8')
+const nginxIngressHelmValues = YAML.parse(nginxIngressHelmValues_file)
+
+let images : {};
+images = helmValues['tezos_k8s_images'] || {};
+
+let imagelist = ["baker_endorser", "chain_initiator", "config_generator",
+	         "rpc_auth", "wait_for_bootstrap", "zerotier"]
+
+for (let image of imagelist) {
+    images[image] = repo.buildAndPushImage("../" + image.replace(/_/g, "-"))
+}
+
+helmValues["tezos_k8s_images"] = images;
```

Comment:
```suggestion
const tezosK8sImages = ["baker_endorser", "chain_initiator", "config_generator",  "rpc_auth", "wait_for_bootstrap", "zerotier"]
const taggedImages = {}

for (const image of tezosK8sImages) {
    taggedImages[image] = repo.buildAndPushImage("../" + image.replace(/_/g, "-"))
}

helmValues["tezos_k8s_images"] = taggedImages;
```

You don't need line 25 bec `helmValues` is coming from the values file that is gend by mkchain. And mkchain doesn't deal with the `tezos_k8s_images`. They are hardcoded in the chart's values.yaml.

Reply:
Your next suggestion overrides this one.